[MIG] partner_contact_lang: backport from v13

yajo opened the pull request:
- Remove stuff related to `active_lang_count`, which doesn't exist in v12.

@Tecnativa TT24503

pedrobaeza committed:
partner_contact_lang: Manage language in contacts (#302)

===========================
Manage language in contacts
===========================

Odoo by default propagate language field to the created contacts from their
form, but it doesn't allow to change it once created.

This module fills this gap, and also provides other facilities for the
contact language management:

* Put the language of the parent company when the contact doesn't have a
  language and this parent company is assigned.
* When the company changes the language, it fills with the same language all
  the contacts that don't have any.

Usage
=====

Go to any partner that is a company and has contacts. Click on one contact
and you will be able to edit the language.

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[MIG] partner_contact_lang: Migration to 9.0

oca-transbot committed:
OCA Transbot updated translations from Transifex

cubells committed:
[MIG] partner_contact_lang: Migration to 10.0

cristinamartinrod committed:
[MIG] partner_contact_lang: Migration to 11.0

pegonzalezsredcor committed:
[IMP] partner_contact_lang: black, isort

pegonzalezsredcor committed:
[MIG] 13.0 partner_contact_lang

Commit:
[MIG] partner_contact_lang: Migration to 12.0

Backport from v13

- Remove stuff related to `active_lang_count`, which doesn't exist in v12.
- Properly pass parent lang to children. Workaround https://github.com/odoo/odoo/issues/53738.

@Tecnativa TT24503

rafaelbn reviewed (changes requested):
Doesn't work. Missing:

1. When you are in the company and you give to add contact, the same language that the parent company has will automatically be put in contact. In any case, the "language" field will not be lost so that it can be modified manually.

2. When creating a contact from contacts, selecting the parent company will change the language to the same as the parent company. Eg, Parent Company has English, when creating contact it puts Spanish, because when selecting the parent company the contact will be automatically with English language.

chienandalu commented:
Looks like those issues are in the v13 as well :confused:

yajo commented:
Seems some problem in odoo itself. It should apply the parent lang by
default
<https://github.com/odoo/odoo/blob/52d932a586d1366351af2a83c72f4bf606386a63/odoo/addons/base/views/res_partner_views.xml#L271>,
but it's not happening. I'd say it's getting the lang from the user context
and not from the active record.

El vie., 26 jun. 2020 a las 12:24, David Vidal (<notifications@github.com>)
escribió:

> Looks like those issues are in the v13 as well 😕
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/OCA/partner-contact/pull/935#issuecomment-650128611>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAHNXDNJ3O6KAX7QASEU6OTRYSAOTANCNFSM4OJGSRRA>
> .
>


-- 
Jairo Llopis

yajo commented:
> When you are in the company and you give to add contact, the same language that the parent company has will automatically be put in contact. In any case, the "language" field will not be lost so that it can be modified manually.

Here's upstream issue. Nothing we can do downstream AFAICS. https://github.com/odoo/odoo/issues/53738

However, if you leave the lang field empty, it will be updated after saving.

> When creating a contact from contacts, selecting the parent company will change the language to the same as the parent company. Eg, Parent Company has English, when creating contact it puts Spanish, because when selecting the parent company the contact will be automatically with English language.

This only happens when current contact (editing/creating in full form) has an empty lang field.

pedrobaeza commented:
> Here's upstream issue. Nothing we can do downstream AFAICS. odoo/odoo#53738
> However, if you leave the lang field empty, it will be updated after saving.

Why don't you create a `lang2` related field and use that one for the default?

yajo commented:
Seems a good workaround. :thinking:

yajo commented:
I added the fix in a separate commit to be able to cherry-pick it in other branches. See now:

![Peek 29-06-2020 09-44](https://user-images.githubusercontent.com/973709/85992800-4ee15780-b9ed-11ea-88a6-ad29b692b4f6.gif)

rafaelbn commented:
@Yajo I tried in runbot but I found the module uninstalled by default. When I tried to install Odoo ask me for a dependency.

See:
![2020-06-29_17-32-10](https://user-images.githubusercontent.com/8736623/86025585-aa800500-ba2e-11ea-9804-91e27d4e38ff.png)

yajo commented:
Solved.

chienandalu commented on partner_contact_lang/models/res_partner.py:
```diff
@@ -0,0 +1,36 @@
+
+    # HACK https://github.com/odoo/odoo/issues/53738
+    # TODO Remove when fixed
+    partner_lang = fields.Selection(related="lang", readonly=True)
```
set a `string` to avoid duplicated label warnings

yajo replied:
Thanks, fixed.

rafaelbn commented:
@Yajo it doesn work in this case, see:

![2020-06-30_12-40-00](https://user-images.githubusercontent.com/8736623/86117285-04380c00-bacf-11ea-9913-fe8a87cec530.png)


> 2. When creating a contact from contacts, selecting the parent company will change the language to the same as the parent company. Eg, Parent Company has English, when creating contact it puts Spanish, because when selecting the parent company the contact will be automatically with English language.

yajo commented:
According to the module README, it doesn't modify the language. It only adds it if the current partner's language is empty. Check this out:

![Peek 30-06-2020 13-38](https://user-images.githubusercontent.com/973709/86126978-1a49ca80-bad7-11ea-8b75-21bac7041616.gif)

I know what you're thinking: when creating a contact, lang should be empty by default, so this happens always. But IMHO that's wrong because not every time you create a contact it belongs to a company, so you could be creating contacts without lang by mistake.

I think this is the best it can be done without fixing one corner case to break other one... :shrug:

rafaelbn commented:
> According to the module README, it doesn't modify the language. It only adds it if the current partner's language is empty. Check this out:

According to the README, the README doesn't have sense at all as it is imposible to create a partner with language empty. Impossible, as Odoo set always a language.

Please FIX the BUG and the REAME with the new text



1.     When you are in the company and you give to add contact, the same language that the parent company has will automatically be put in contact. In any case, the "language" field will not be lost so that it can be modified manually.
2.     When creating a contact from contacts, selecting the parent company will change the language to the same as the parent company. Eg, Parent Company has English, when creating contact it puts Spanish, because when selecting the parent company the contact will be automatically with English language.

pedrobaeza committed:
[IMP] partner_contact_lang: Change language when assigning parent if virtual record

If the record hasn't been created yet, we can assume that the language set currently
is the default one, so we can switch it to the parent one.

pedrobaeza reviewed (approved):
Fixed the parent language assignation when the record is still virtual

rafaelbn reviewed (approved):
:+1:

pedrobaeza commented:
/ocabot merge nobump

pedrobaeza commented:
Cherry-picked extra commit to v13 in d691178d339d1b81a8a1e718121fbd70a6a0cb30
